cartographer/mapping/internal/2d/local_trajectory_builder_2d.cc

```diff
@@ -88,7 +89,7 @@ std::unique_ptr<transform::Rigid2d> LocalTrajectoryBuilder2D::ScanMatch(
                             filtered_gravity_aligned_point_cloud,
                             *matching_submap->grid(), pose_observation.get(),
                             &summary);
-  if (pose_observation) {
+  if (pose_observation && metrics::MetricsRegistered()) {
```

Comment:
@gaschler An example in the current code - makes it obvious that the following code block is only for metrics.

Reply:
I'd prefer it for instrumentation code that is  a bit more involved than just a single Observe() etc.

Reply:
We already have a lot of code paths, so I prefer instrumentation to always execute.
cschuet@ pfion2a@ What do you think?